tests/test_domains_config.py

```diff
@@ -168,6 +168,21 @@ def test_get_volume_config_error(domains_config):
     assert namespace == None
 
 
+def test_volume_config_hostname_worker_id_cache(domains_config):
+    dc = domains_config
+    worker_id = 'worker1'
+    hostname = 'example007.com'
+    dc._cwm_api_volume_configs['hostname:{}'.format(hostname)] = {
+        'instanceId': worker_id, 'minio_extra_configs': {'hostnames': [{'hostname': hostname}]}
+    }
+    volume_config = dc.get_cwm_api_volume_config(hostname=hostname)
+    assert volume_config.id == worker_id
+    assert domains_config.keys.volume_config_hostname_worker_id.get(hostname).decode() == worker_id
```

Comment:
you need to verify that the volume config is received from cache, currently this test only verifies that the redis key is set

you ca verify that by deleting or setting an invalid value in `dc._cwm_api_volume_configs['hostname:{}'.format(hostname)]` then making another call to `get_cwm_api_volume_config` - but because the value is in cache it will still return the correct value from the cache

Reply:
Currently, the `worker_id` is cached against `hostname`, not the complete `VolumeConfig` object.
The cached `worker_id` will be returned in this case if accessed from the cache.
Should I change it and cache the `VolumeConfig` to test the reset/delete scenario?

For example:

**domain_config.py**

```python
        if domains_config and request_hostname and self.id is not None:
            domains_config.keys.volume_config_hostname_worker_id.set(request_hostname, json.dumps(request_data))
```

**tests/test_domains_config.py**

```python
def test_volume_config_hostname_worker_id_cache(domains_config):
    dc = domains_config
    worker_id = 'worker1'
    hostname = 'example007.com'
    dc._cwm_api_volume_configs['hostname:{}'.format(hostname)] = {
        'instanceId': worker_id, 'minio_extra_configs': {'hostnames': [{'hostname': hostname}]}
    }
    volume_config_1 = dc.get_cwm_api_volume_config(hostname=hostname)
    assert domains_config.keys.volume_config_hostname_worker_id.get(hostname)

    # reset volume config and try to get from cache
    dc._cwm_api_volume_configs['hostname:{}'.format(hostname)] = {}
    volume_config_2 = dc.get_cwm_api_volume_config(hostname=hostname)
    assert domains_config.keys.volume_config_hostname_worker_id.get(hostname)

    assert volume_config_1 != volume_config_2
    assert volume_config_1.id == volume_config_2.id
    assert volume_config_1.hostnames == volume_config_2.hostnames

    dc.del_worker_hostname_keys(hostname)
    assert domains_config.keys.volume_config_hostname_worker_id.get(hostname) == None
```

Reply:
no, you should only cache the worker_id, the implementation seems fine, just need to write a test for it